tools/topology/topology2/development/cavs-nocodec-asrc.conf

```diff
@@ -0,0 +1,758 @@
+				in_ch_map	$CHANNEL_MAP_3_POINT_1
+				out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				out_ch_map	$CHANNEL_MAP_3_POINT_1
+			}
+		}
+	}
+
+	{
+		index 	4
+		Object.Widget.host-copier.1 {
+			stream_name "SSP0 Capture"
+			pcm_id		$SSP0_PCM_ID
+		}
+	}
+
+	{
+		index 	8
+		Object.Widget.host-copier.1 {
+			stream_name "SSP1 Capture"
+			pcm_id		$SSP1_PCM_ID
+		}
+	}
+
+	{
+		index 	12
+		Object.Widget.host-copier.1 {
+			stream_name "SSP2 Capture"
+			pcm_id		$SSP2_PCM_ID
+			<include/components/asrc_format_sxx_to_s32_convert.conf>
+		}
+	}
+]
+Object.Base.route [
+	{
+		source $DMIC0_DAI_COPIER
+		sink "asrc.1.1"
+	}
+
+	{
+		source "asrc.1.1"
+		sink	"host-copier.3.capture"
+	}
+
+	{
+		source	"dai-copier.SSP.NoCodec-0.capture"
+		sink	"host-copier.$SSP0_PCM_ID.capture"
+	}
+
+	{
+		source	"dai-copier.SSP.NoCodec-1.capture"
+		sink	"asrc.7.1"
+	}
+
+	{
+		source	"asrc.7.1"
+		sink	"host-copier.$SSP1_PCM_ID.capture"
+	}
+
+	{
+		source	"dai-copier.SSP.NoCodec-2.capture"
+		sink	"asrc.11.1"
+	}
+
+	{
+		source	"asrc.11.1"
+		sink	"host-copier.$SSP2_PCM_ID.capture"
+	}
+
+
+]
+
+Object.PCM.pcm [
+	{
+		name	"ASRC Capture 3"
+		id 3
+		direction	"capture"
+		Object.Base.fe_dai."ASRC Capture 3" {}
+
+		Object.PCM.pcm_caps."capture" {
+			name "ASRC Capture 3"
+			# only 32-bit capture supported now
+			formats 'S32_LE'
+			channels_min $NUM_DMICS
+			channels_max $NUM_DMICS
+		}
+	}
+]
+
+
+Object.PCM.pcm [
+	{
+		name	"$SSP0_PCM_NAME"
+		id $SSP0_PCM_ID
+		direction	"duplex"
+		Object.Base.fe_dai.1 {
+			name	"$SSP0_PCM_NAME"
+		}
+
+		Object.PCM.pcm_caps.1 {
+			direction	"playback"
+			name "SSP0 Playback"
+			formats 'S16_LE,S24_LE,S32_LE'
+		}
+
+		Object.PCM.pcm_caps.2 {
+			direction	"capture"
+			name "SSP0 Capture"
+			formats 'S16_LE,S24_LE,S32_LE'
+		}
+	}
+
+	{
+		name	"$SSP1_PCM_NAME"
+		id $SSP1_PCM_ID
+		direction	"duplex"
+		Object.Base.fe_dai.1 {
+			name	"$SSP1_PCM_NAME"
+		}
+
+		Object.PCM.pcm_caps.1 {
+			direction	"playback"
+			name "SSP1 Playback"
+			formats 'S16_LE,S24_LE,S32_LE'
+		}
+
+		Object.PCM.pcm_caps.2 {
+			direction	"capture"
+			name "SSP1 Capture"
+			formats 'S16_LE,S24_LE,S32_LE'
+		}
+	}
+
+	{
+		name	"$SSP2_PCM_NAME"
+		id $SSP2_PCM_ID
+		direction	"duplex"
+		Object.Base.fe_dai.1 {
+			name	"$SSP2_PCM_NAME"
+		}
+
+		Object.PCM.pcm_caps.1 {
+			direction	"playback"
+			name "SSP2 Playback"
+			formats 'S16_LE,S24_LE,S32_LE'
+		}
+
+		Object.PCM.pcm_caps.2 {
+			direction	"capture"
+			name "SSP2 Capture"
+			formats 'S16_LE,S24_LE,S32_LE'
+		}
+	}
+]
+
+
+
+# PCM0 --> ASRC--> SSP0
+
+# PCM1 --> ASRC--> SSP1
+
+# PCM2--> SSP2
+
+Object.Pipeline.asrc-dai-copier-playback [
+	{
+		index	5
+		Object.Widget.dai-copier.1 {
+			dai_index 0
+			dai_type "SSP"
+			copier_type "SSP"
+			stream_name "NoCodec-0"
+			node_type $I2S_LINK_OUTPUT_CLASS
+			num_input_pins 1
+			<include/components/asrc_format_sxx_to_s32_convert.conf>
+		}
+		Object.Widget.asrc."1" {
+			rate_in 48000
+			rate_out 44100
```

Comment:
Is they playback DAI really 44.1 kHz?

Reply:
sorry, used don't know that SSP can not do sample convert.